### PR TITLE
Saving and position columns bug fixes and changes

### PR DESCRIPTION
**Pull request by dannyparsons**
1. Fixed functionality so that when replacing an existing column, it stays in same position by default and also added parameter to the R methods to allow it to be moved when replaced (a control for this can be added to the positioning sub dialog in a separate PR)
2. Fixed bug in calculator where the `before` parameter, and therefore positioning a new column, was changed when reopening the dialog
3. Added `strAdjacentColumn` as a parameter to assign to methods as this is now part of the saving options

@africanmathsinitiative/developers This is ready for the review. 1. needs to be reviewed across dialogs that use the save control, please test carefully as this edits core parts of saving columns, and 2. only needs to be checked for the calculator dialog

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Patowhiz You've done quite a bit of work in this area, please could you peer review? thanks

**Review by rdstern — APPROVED** [review]
Good

File: instat/static/InstatObject/R/instat_object_R6.R [hunk 1]
```diff
@@ -478,9 +478,9 @@ DataBook$set("public", "set_metadata_changed", function(data_name = "", new_val)
...
-  if(missing(use_col_name_as_prefix)) self$get_data_objects(data_name)$add_columns_to_data(col_name, col_data, hidden = hidden, before = before, adjacent_column = adjacent_column, num_cols = num_cols, require_correct_length = require_correct_length)
-  else self$get_data_objects(data_name)$add_columns_to_data(col_name, col_data, use_col_name_as_prefix = use_col_name_as_prefix, hidden = hidden, before = before, adjacent_column = adjacent_column, num_cols = num_cols, require_correct_length = require_correct_length)
+DataBook$set("public", "add_columns_to_data", function(data_name, col_name = "", col_data, use_col_name_as_prefix = FALSE, hidden = FALSE, before, adjacent_column, num_cols, require_correct_length = TRUE, keep_existing_position = TRUE) {
+  if(missing(use_col_name_as_prefix)) self$get_data_objects(data_name)$add_columns_to_data(col_name, col_data, hidden = hidden, before = before, adjacent_column = adjacent_column, num_cols = num_cols, require_correct_length = require_correct_length, keep_existing_position = keep_existing_position)
```
Comment (Patowhiz) [inline instat/static/InstatObject/R/instat_object_R6.R:482]: @dannyparsons will parameter `use_col_name_as_prefix` ever be missing? I can see it has a default value `False`.

File: instat/clsRCodeStructure.vb [hunk 2]
```diff
@@ -250,8 +250,7 @@ Public Class RCodeStructure
...
-    Public Sub SetAssignTo(strTemp As String, Optional strTempDataframe As String = "", Optional strTempColumn As String = "", Optional strTempModel As String = "", Optional strTempGraph As String = "", Optional strTempSurv As String = "", Optional strTempTable As String = "", Optional bAssignToIsPrefix As Boolean = False, Optional bAssignToColumnWithoutNames As Boolean = False, Optional bInsertColumnBefore As Boolean = False, Optional bRequireCorrectLength As Boolean = True, Optional bDataFrameList As Boolean = False, Optional strDataFrameNames As String = "")
-
+    Public Sub SetAssignTo(strTemp As String, Optional strTempDataframe As String = "", Optional strTempColumn As String = "", Optional strTempModel As String = "", Optional strTempGraph As String = "", Optional strTempSurv As String = "", Optional strTempTable As String = "", Optional bAssignToIsPrefix As Boolean = False, Optional bAssignToColumnWithoutNames As Boolean = False, Optional bInsertColumnBefore As Boolean = False, Optional bRequireCorrectLength As Boolean = True, Optional bDataFrameList As Boolean = False, Optional strDataFrameNames As String = "", Optional strAdjacentColumn As String = "")
```
Comment (Patowhiz) [inline instat/clsRCodeStructure.vb:253]: Would be good to add the parameter added `strAdjacentColumn` to the list of parameters in the comment.

Comment (Patowhiz) [reply instat/clsRCodeStructure.vb:253]: @dannyparsons probably not related to the added functionality.
In the same `SetAssignTo()` I have noticed that some optional parameters are always overwriting the initially set property values while others don't.

Shouldn't we have the same consistency? As a developer using the object, I would think that if I don't define the value of the optional parameter then the internal value that it's meant to assign to, won't be overwritten. 

Alternatively we could include a comment to the optional values that are always used in the subroutine regardless of whether they are used or not, just to be clear.

**Review by Patowhiz — COMMENTED** [review]
This looks good. Tested it across multiple dialogs that use the ucrSave control. `ucrSave` control will need to be modified to enable positioning of existing columns and take advantage of the new parameter added. Issue #6002 will also need to be reopened to reimplement the functionality requested.

Just a few questions to @dannyparsons.

**Comment by Patowhiz** [comment]
@dannyparsons @lloyddewit, on a separate note, now that @dannyparsons has worked on the `SetAssignTo()` subroutine, I think in line 218 `If bUseBaseFunction Then 'TODO SJL 07/04/20 Use ElseIf?` of `clsRSyntax.vb`.
The todo could now be done by adding the `ElseIf` as recommended

File: instat/static/InstatObject/R/instat_object_R6.R (hunk 1 above)
Comment (dannyparsons) [reply instat/static/InstatObject/R/instat_object_R6.R]: `missing(use_col_name_as_prefix)` is still `TRUE` if it is not specified by the user, even if it has a default. But I don't think that's relevant here and there isn't a need for the `if` `else` anymore so I've removed that.

File: instat/clsRCodeStructure.vb (hunk 2 above)
Comment (dannyparsons) [reply instat/clsRCodeStructure.vb:254]: Added the comment.

Yes we probably don't want to overwrite unless the user has specified. We should probably have all the defaults as `Nothing` or `Null` and only change if not. I would want to first check if any uses of this rely on the existing implementation before changing. So I think this should be a separate issue/PR to do this.

**Comment by dannyparsons** [comment]
Thanks, done and addressed the comments.

**Comment by Patowhiz** [comment]
@lloyddewit this looks good to me.

**Pull request merged**